firefox_ui_tests/functional/locationbar/test_star_in_autocomplete.py

```diff
@@ -19,24 +21,26 @@ class TestStarInAutocomplete(FirefoxTestCase):
     def setUp(self):
         FirefoxTestCase.setUp(self)
 
-        self.clear_history()
+        self.clean_up_state()
 
         # Location bar suggests 'History and Bookmarks'
         self.prefs.set_pref(self.PREF_LOCATION_BAR_SUGGEST, 0)
 
     def tearDown(self):
-        FirefoxTestCase.tearDown(self)
-
-        self.prefs.reset_pref(self.PREF_LOCATION_BAR_SUGGEST)
-
-    def clear_history(self):
+        # Close the autocomplete results
+        try:
+            self.browser.navbar.locationbar.autocomplete_results.close()
+            self.places.restore_default_bookmarks()
+        except NoSuchElementException:
+            pass
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def clean_up_state(self):
         with self.marionette.using_context('content'):
             self.marionette.navigate('about:blank')
 
-        self.marionette.execute_script('''
-            let count = gBrowser.sessionHistory.count;
-            gBrowser.sessionHistory.PurgeHistory(count);
-        ''')
+        self.places.remove_all_history()
```

Comment:
You can get rid of this extra method, and put those lines into setUp() directly.